src/python/pants/option/parser.py

```diff
@@ -311,11 +311,9 @@ def add_flag_val(v: Optional[Union[int, float, bool, str]]) -> None:
                 # Reraise a new exception with context on the option being processed at the time of error.
                 # Note that other exception types can be raised here that are caught by ParseError (e.g.
                 # BooleanConversionError), hence we reference the original exception type as type(e).
+                args_str = ", ".join(args)
                 raise type(e)(
-                    "Error computing value for {} in {} (may also be from PANTS_* environment variables)."
-                    "\nCaused by:\n{}".format(
-                        ", ".join(args), self._scope_str(), traceback.format_exc()
-                    )
+                    f"Error computing value for {args_str} in {self._scope_str()} (may also be from PANTS_* environment variables).\nCaused by:\n{traceback.format_exc()}"
```

Comment:
The long line is a step back in usability I think, especially since the line contains a newline.